src/ast_dumper.cpp

```diff
@@ -143,27 +154,34 @@ void AstDumper::Visit(const BinaryExprNode& bin_expr) {
             << ExprTypeToCString(bin_expr.type) << std::endl;
 }
 
-/// @brief Dispatch the concrete binary expressions to the parent
-/// `BinaryExprNode`.
-/// @param classname A subclass of `BinaryExprNode`.
-#define DISPATCH_TO_VISIT_BINARY_EXPR(classname) \
+/// @brief Dispatch the concrete binary or unary expressions to the parent
+/// `BinaryExprNode` or `UnaryExprNode`.
+/// @param classname A subclass of `BinaryExprNode` or `UnaryExprNode`.
+#define DISPATCH_TO_VISIT_EXPR(parentname, classname) \
   void AstDumper::Visit(const classname& expr) { \
-    Visit(static_cast<const BinaryExprNode&>(expr)); \
+    Visit(static_cast<const parentname&>(expr)); \
   }
 
-DISPATCH_TO_VISIT_BINARY_EXPR(PlusExprNode);
-DISPATCH_TO_VISIT_BINARY_EXPR(SubExprNode);
-DISPATCH_TO_VISIT_BINARY_EXPR(MulExprNode);
-DISPATCH_TO_VISIT_BINARY_EXPR(DivExprNode);
-DISPATCH_TO_VISIT_BINARY_EXPR(ModExprNode);
-DISPATCH_TO_VISIT_BINARY_EXPR(GreaterThanExprNode);
-DISPATCH_TO_VISIT_BINARY_EXPR(GreaterThanOrEqualToExprNode);
-DISPATCH_TO_VISIT_BINARY_EXPR(LessThanExprNode);
-DISPATCH_TO_VISIT_BINARY_EXPR(LessThanOrEqualToExprNode);
-DISPATCH_TO_VISIT_BINARY_EXPR(EqualToExprNode);
-DISPATCH_TO_VISIT_BINARY_EXPR(NotEqualToExprNode);
-
-#undef DISPATCH_TO_VISIT_BINARY_EXPR
+DISPATCH_TO_VISIT_EXPR(BinaryExprNode, PlusExprNode);
+DISPATCH_TO_VISIT_EXPR(BinaryExprNode, SubExprNode);
+DISPATCH_TO_VISIT_EXPR(BinaryExprNode, MulExprNode);
+DISPATCH_TO_VISIT_EXPR(BinaryExprNode, DivExprNode);
+DISPATCH_TO_VISIT_EXPR(BinaryExprNode, ModExprNode);
+DISPATCH_TO_VISIT_EXPR(BinaryExprNode, GreaterThanExprNode);
+DISPATCH_TO_VISIT_EXPR(BinaryExprNode, GreaterThanOrEqualToExprNode);
+DISPATCH_TO_VISIT_EXPR(BinaryExprNode, LessThanExprNode);
+DISPATCH_TO_VISIT_EXPR(BinaryExprNode, LessThanOrEqualToExprNode);
+DISPATCH_TO_VISIT_EXPR(BinaryExprNode, EqualToExprNode);
+DISPATCH_TO_VISIT_EXPR(BinaryExprNode, NotEqualToExprNode);
```

Comment:
```suggestion
DISPATCH_TO_VISIT_EXPR(BinaryExprNode, NotEqualToExprNode);

```
Add a vertical separation between binary and unary expressions.